src/main/kotlin/com/cjbooms/fabrikt/model/KotlinGenModels.kt

```diff
@@ -68,3 +97,48 @@ private fun <T : GeneratedType> groupClasses(allModels: Collection<T>): Map<T, L
         .toMap()
     return sealedClasses.plus(otherClasses)
 }
+
+/**
+ * The IncomingParameter class is intended to represent a given name and type
+ * of an incoming request, be it either a header, url param, path param, or body
+ */
+sealed class IncomingParameter(val oasName: String, val type: TypeName) {
+    val name: String = oasName.toKotlinParameterName()
+    fun toParameterSpecBuilder(): ParameterSpec.Builder =
+        ParameterSpec.builder(name, type)
+}
+
+class BodyParameter(oasName: String, type: TypeName, val schema: Schema) : IncomingParameter(oasName, type)
+
+class RequestParameter(
```

Comment:
I'm a bit confused by `RequestParameter` and `BodyParameter`. `RequestParameter` has a `ParameterLocation` which could be `object BodyParam : ParameterLocation()`

Reply:
yeah, it looks like a factory pattern with the wrong inheritance. Let me have a look to see if we can easily re-factor it